File: easy/1752_check_if_array_is_sorted_and_rotated/solution.py
from typing import List


class Solution:
    def check(self, nums: List[int]) -> bool:
        nums_len = len(nums)

        i = 0

        while i < nums_len - 1:
            if nums[i] > nums[i + 1]:
                break

            i += 1

        if i == nums_len - 1:
            return True

        i += 1

        while i < nums_len - 1:
            if nums[i] > nums[i + 1]:
                return False

            i += 1

        return nums[0] >= nums[-1]

File: easy/1752_check_if_array_is_sorted_and_rotated/solution.cpp
#include <vector>

class Solution {
public:
  bool check(std::vector<int> &nums) {
    const int numsLen = nums.size();

    int i;
    for (i = 1; i < numsLen && nums[i - 1] <= nums[i]; ++i) {
    }

    if (i == numsLen)
      return true;

    for (; i < numsLen - 1; ++i)
      if (nums[i] > nums[i + 1])
        return false;

    return nums[0] >= nums[numsLen - 1];
  }
};
